src/Servers/HttpSys/src/FeatureContext.cs

```diff
@@ -316,7 +316,17 @@ X509Certificate2 ITlsConnectionFeature.ClientCertificate
             {
                 if (IsNotInitialized(Fields.ClientCertificate))
                 {
-                    _clientCert = Request.GetClientCertificateAsync().Result; // TODO: Sync;
+                    var mode = _requestContext.Server.Options.ClientCertificateMode;
+                    if (mode == ClientCertificateMode.AllowCertificate)
+                    {
+                        _clientCert = Request.ClientCertificate;
+                    }
+                    else if (mode == ClientCertificateMode.DynamicCertificate)
+                    {
+                        _clientCert = Request.GetClientCertificateAsync().Result; // TODO: Sync;
```

Comment:
Should we remove the comment? Or change it to "TODO: Make non-default"? Or file a new issue to change the default?

The current TODO doesn't indicate what action we want to take. We can't make this getter return `Task<X509Certificate2>` without breaking people, so it doesn't seem that making the DynamicCertificate mode async is an option.

Reply:
Yeah, the comment is definitely confusing.

Reply:
That TODO was in the old code and referred to the sync over async. I'll clarify.